### PR TITLE
Enhancement rest implement delete resource relation

### PR DESCRIPTION
**Pull request by RetGal**
Implements #361

### CODE REVIEW

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java [hunk 1]
```diff
@@ -164,11 +153,68 @@ public Response addRelation(@PathParam("slaveResourceGroupName") String slaveGro
...
+    /**
+     * Removes a new ResourceRelation
+     *
+     * @param relationName A String representing the identifier of a Relation or the name of the related Resource to be removed
+     */
+    @Path("/{relationName}")
+    @DELETE
+    @ApiOperation(value = "Remove a Relation from a specific Release", notes = "RelationName may be the identifier of a Relation or the name of the related Resource")
+    public Response removeRelation(@PathParam("relationName") String relationName) throws ValidationException {
+        if (StringUtils.isEmpty(relationName)) {
```
Comment (yvespp) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java]: Duplicate code of line 149?

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java [hunk 2]
```diff
@@ -164,11 +153,68 @@ public Response addRelation(@PathParam("slaveResourceGroupName") String slaveGro
...
+            ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithProvidedRelations(resourceGroupName, releaseName);
+            relations = resource.getProvidedMasterRelations();
+        }
+        return relations;
+    }
+
+    private void removeMatchingRelation(Collection<? extends AbstractResourceRelationEntity> relations, String relationName)
+            throws ForeignableOwnerViolationException, ResourceNotFoundException, ElementAlreadyExistsException {
+        for (AbstractResourceRelationEntity relation : relations) {
+            if (isMatchingRelationName(relation, relationName)) {
```
Comment (yvespp) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java]: What happens if no relations matches? Should return an error.

File: AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java [hunk 3]
```diff
@@ -164,11 +153,68 @@ public Response addRelation(@PathParam("slaveResourceGroupName") String slaveGro
...
+            ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithConsumedRelations(resourceGroupName, releaseName);
+            relations = resource.getConsumedMasterRelations();
+        } else if (resourceType.toLowerCase().equals("provided")) {
+            ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithProvidedRelations(resourceGroupName, releaseName);
+            relations = resource.getProvidedMasterRelations();
+        }
+        return relations;
+    }
+
+    private void removeMatchingRelation(Collection<? extends AbstractResourceRelationEntity> relations, String relationName)
```
Comment (yvespp) [inline AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java]: Shouldn't this be done in the backend/business package.

**Comment by yvespp** [comment]
Also opened #388 and #387

**Comment by RetGal** [comment]
@yvespp Shall I address #388 and #387 as well?

**Comment by yvespp** [comment]
@RetGal yes but better in a new pull request.

**Pull request merged**